src/swap/SwapScreen.tsx

```diff
@@ -429,28 +430,58 @@ export function SwapScreen({ route }: Props) {
   }
 
   const handleSelectToken = (selectedToken: TokenBalance) => {
-    if (selectedToken && selectingToken) {
-      ValoraAnalytics.track(SwapEvents.swap_screen_confirm_token, {
-        fieldType: selectingToken,
-        tokenSymbol: selectedToken.symbol,
-        tokenId: selectedToken.tokenId,
-        tokenNetworkId: selectedToken.networkId,
-      })
-    }
-
+    let switchedToNetworkId: NetworkId | null = null
+    let newFromToken: TokenBalance | undefined = fromToken
+    let newToToken: TokenBalance | undefined = toToken
     if (
       (selectingToken === Field.FROM && toToken?.tokenId === selectedToken.tokenId) ||
       (selectingToken === Field.TO && fromToken?.tokenId === selectedToken.tokenId)
     ) {
-      setFromToken(toToken)
-      setToToken(fromToken)
+      newFromToken = toToken
+      newToToken = fromToken
+      setFromToken(newFromToken)
+      setToToken(newToToken)
     } else if (selectingToken === Field.FROM) {
-      setFromToken(selectedToken)
+      newFromToken = selectedToken
+      setFromToken(newFromToken)
+      switchedToNetworkId =
+        toToken && toToken.networkId !== newFromToken.networkId ? newFromToken.networkId : null
+      setSwitchedToNetworkId(switchedToNetworkId)
+      if (switchedToNetworkId) {
+        // reset the toToken if the user is switching networks
+        setToToken(undefined)
+      }
     } else if (selectingToken === Field.TO) {
-      setToToken(selectedToken)
+      newToToken = selectedToken
+      setToToken(newToToken)
+      switchedToNetworkId =
+        fromToken && fromToken.networkId !== newToToken.networkId ? newToToken.networkId : null
+      setSwitchedToNetworkId(switchedToNetworkId)
+      if (switchedToNetworkId) {
+        // reset the fromToken if the user is switching networks
+        setFromToken(undefined)
+      }
```

Comment:
It may look like we have duplicated similar logic, I tried to make it more concise but then I thought it was making it complicated to understand.

But let me know if you have a better idea.